File: 2032-largest-odd-number-in-string/largest-odd-number-in-string.java
class Solution {
    public String largestOddNumber(String num) {
        for(int right=num.length()-1; right>=0; right--) {
            if((num.charAt(right) - '0') % 2 != 0) {
                return num.substring(0, right+1);
            }
        }
        return "";
    }
}

File: 2032-largest-odd-number-in-string/largest-odd-number-in-string.js
/**
 * @param {string} num
 * @return {string}
 */
var largestOddNumber = function(num) {
    let right = -1;
    for(let i=num.length-1; i>=0; i--) {
        if(num[i] % 2 !== 0) {
            right = i;
            break;
        }
    }
    let ans = '';
    for(let i=0; i<=right; i++) {
        ans += num[i];
    }
    return ans;
};